src/psyclone/dynamo0p3.py

```diff
@@ -2364,7 +2364,8 @@ def __init__(self, alg_invocation, idx):
                 for scalar in loop.args_filter(
                         arg_types=VALID_SCALAR_NAMES,
                         arg_accesses=VALID_REDUCTION_NAMES, unique=True):
-                    if scalar.type.lower() == "gh_integer":
+                    if scalar.type.lower() == "gh_integer" and \
```

Comment:
I think this change is unnecessary since `args_filter` is being used to only return scalar arguments that are used in reductions (`arg_accesses=VALID_REDUCTION_NAMES`). Therefore if we've got an integer scalar at this point then we know that it's not valid.

Reply:
I suspected this. Removed.